packages/evm/src/interpreter.ts

```diff
@@ -234,10 +238,30 @@ export class Interpreter {
     let doJumpAnalysis = true
     // Iterate through the given ops until something breaks or we hit STOP
     while (this._runState.programCounter < this._runState.code.length) {
+      const programCounter = this._runState.programCounter
       let opCode: number
       let opCodeObj: OpcodeMapEntry
       if (doJumpAnalysis) {
-        opCode = this._runState.code[this._runState.programCounter]
+        opCode = this._runState.code[programCounter]
```

Comment:
This is inside `doJumpAnalysis`, is this correct? So this 0xfe check is only done if we have not yet done the jumpdest analysis :thinking:

Reply:
I think this is problematic if we have already done the jumpdest analysis (so, after we have done a JUMP opcode). Then `doJumpAnalysis` is `false`, so we dont do the invalid check. However, if we then hit an invalid opcode we do not know if this was the real invalid opcode or if we dont have this chunk.

(Also, this assumes that if `this._runState.code[programCounter]` returns 0xfe if we are outside the code chunk - is this correct and is this done somewhere? I guess I'm missing something at this point :) )

Reply:
ahh yes this needs to be outside

Reply:
> (Also, this assumes that if this._runState.code[programCounter] returns 0xfe if we are outside the code chunk - is this correct and is this done somewhere? I guess I'm missing something at this point :)

yes code array is intialized with `0xfe` and then code chunks provided are filled in